packages/config/src/memory/memory.config.ts

```diff
@@ -140,8 +177,14 @@ export class ConfigProviderMemory extends BasemapsConfigProvider {
       removeUndefined(existing);
       this.put(existing);
     }
-    // TODO this overwrites existing layers
-    existing.layers[0][i.projection] = i.id;
+    // The latest imagery overwrite the earlier ones.
+    const existingImageryId = existing.layers[0][i.projection];
+    if (existingImageryId) {
+      existing.layers[0][i.projection] = this.findLatestId(existingImageryId, i.id);
```

Comment:
do we have a logger and can we log that this is being overwritten?

If its not loggable maybe we could store these as `this.duplicateImagery = [];`